From: Egbert Eich <eich@suse.de>
Date: Tue Jan 28 13:08:12 2014 +0100
Subject: drm/mgag200: Add additional limits for certain G200 variants
Patch-mainline: Not yet, upstreaming planned after more testing.

According to the use UMS X.Org driver G200 WB chips don't support
doublescan

Signed-off-by: Egbert Eich <eich@suse.de>
Signed-off-by: Egbert Eich <eich@suse.com>
---
 drivers/gpu/drm/mgag200/mgag200_mode.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)
diff --git a/drivers/gpu/drm/mgag200/mgag200_mode.c b/drivers/gpu/drm/mgag200/mgag200_mode.c
index 844db9b..ff105b6 100644
--- a/drivers/gpu/drm/mgag200/mgag200_mode.c
+++ b/drivers/gpu/drm/mgag200/mgag200_mode.c
@@ -1704,6 +1704,7 @@ static struct drm_connector *mga_vga_init(struct drm_device *dev)
 {
 	struct drm_connector *connector;
 	struct mga_connector *mga_connector;
+	struct mga_device *mdev = dev->dev_private;
 
 	mga_connector = kzalloc(sizeof(struct mga_connector), GFP_KERNEL);
 	if (!mga_connector)
@@ -1712,7 +1713,7 @@ static struct drm_connector *mga_vga_init(struct drm_device *dev)
 	connector = &mga_connector->base;
 
 	connector->interlace_allowed = true;
-	connector->doublescan_allowed = true;
+	connector->doublescan_allowed = (mdev->type == G200_WB) ? false : true;
 
 	drm_connector_init(dev, connector,
 			   &mga_vga_connector_funcs, DRM_MODE_CONNECTOR_VGA);
